horizon/enforcer/api.py

```diff
@@ -246,6 +301,102 @@ async def is_allowed_url(
         )
         return await is_allowed(request, allowed_query, x_permit_sdk_language)
 
+    @router.post(
+        "/allowed/all-tenants",
+        response_model=AllTenantsAuthorizationResult,
+        status_code=status.HTTP_200_OK,
+        response_model_exclude_none=True,
+        dependencies=[Depends(enforce_pdp_token)],
+    )
+    async def is_allowed_all_tenants(
+        request: Request,
+        query: AuthorizationQuery,
+        x_permit_sdk_language: Optional[str] = Depends(notify_seen_sdk),
+    ):
+        fallback_response = dict(result=dict(allow=[], debug="OPA not responding"))
+        response = await is_allowed_with_fallback(
+            query, request, ALL_TENANTS_POLICY_PACKAGE, fallback_response
+        )
+        log_query_result(query, response)
+        try:
+            raw_result = json.loads(response.body).get("result", {})
+            processed_query = (
+                get_v1_processed_query(raw_result)
+                or get_v2_processed_query(raw_result)
+                or {}
+            )
+            allowed_tenants_result = []
+            debug_result = []
+            for allowed_tenant in raw_result.get("allowed_tenants", []):
```

Comment:
Why not let Pydantic do this processing?

Reply:
changed